### ISSUE TITLE
Different pre-processing times with PyTorch Hub and detect.py

### ISSUE THREAD

**Issue by DP1701** [ultralytics/yolov5#11370]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello, 

why is the pre-procssing time with PyTorch Hub higher than with detect.py?

I use a custom yolov5 model and an image of size 1280x720px

PyTorch Hub:
<img width="653" alt="2" src="https://user-images.githubusercontent.com/68444979/232334071-66201337-e89e-40ac-aa38-7597a583fe1b.png">

detect.py
<img width="823" alt="1" src="https://user-images.githubusercontent.com/68444979/232334086-a6a7faa2-b4d2-4b97-8736-fae2e9ed7648.png">

Another question: when given the size of 1280, 720 becomes 736. Which technique is used to fill the missing pixels? Zero padding?

**Comment by DP1701** [comment]
@glenn-jocher Could you perhaps briefly answer the above questions? I would be very grateful.

**Comment by glenn-jocher** [comment]
@DP1701 certainly! The pre-processing time with PyTorch Hub may be slightly longer because it pre-processes the image on the CPU instead of the GPU. Additionally, the pre-processing times may vary as the PyTorch Hub is a pre-trained model and may perform more operations than necessary for inference. 

Regarding the second question, when the input image size is not divisible by a certain factor, some architectures including YOLOv5 will pad the image with zeros. This is done to ensure that the outputs have the correct dimensions, and the padding is added symmetrically around the image.

**Comment by DP1701** [comment]
@glenn-jocher thank you for the fast answer! If I understand correctly, PyTorch Hub is not the best idea to run inference in real time on a system. I am looking for a way to run inference optimally using simple methods. The hardware for the system is not fixed yet, but I wanted to try it out in some way.

**Comment by glenn-jocher** [comment]
@DP1701 PyTorch Hub is still a good option for running inference and can achieve real-time performance depending on the hardware. However, if you're looking for a more optimized solution, I recommend trying out other methods such as using the PyTorch/CUDA backend, or optimizing the model architecture for your specific use case (e.g. reducing the number of layers or channels). You could also try running the model on a dedicated inference hardware like an NVIDIA Jetson device or a Google Coral board which has specialized hardware for accelerating inference.
